pkg/stdlib/sql.go

```diff
@@ -0,0 +1,377 @@
+	d.configMutex.Unlock()
+}
+
+type driverConnector struct {
+	driver *Driver
+	name   string
+}
+
+func (dc *driverConnector) Connect(ctx context.Context) (driver.Conn, error) {
+	dc.driver.configMutex.Lock()
+	connConfig := dc.driver.configs[dc.name]
+	dc.driver.configMutex.Unlock()
+	return connConfig, nil
+}
+
+func (dc *driverConnector) Driver() driver.Driver {
+	return dc.driver
+}
+
+type Conn struct {
+	conn    client.ImmuClient
+	options *client.Options
+	Token   string
+}
+
+// Conn returns the underlying client.ImmuClient
+func (c *Conn) GetImmuClient() client.ImmuClient {
+	return c.conn
+}
+
+func (c *Conn) GetToken() string {
+	return c.Token
+}
+
+func (c *Conn) Prepare(query string) (driver.Stmt, error) {
+	return nil, ErrNotImplemented
+}
+
+func (c *Conn) PrepareContext(ctx context.Context, query string) (driver.Stmt, error) {
+	return nil, ErrNotImplemented
+}
+
+func (c *Conn) Close() error {
+	return c.conn.Disconnect()
+}
+
+func (c *Conn) Begin() (driver.Tx, error) {
+	return nil, ErrNotImplemented
+}
+
+func (c *Conn) BeginTx(ctx context.Context, opts driver.TxOptions) (driver.Tx, error) {
+	return nil, ErrNotImplemented
+}
+
+func (c *Conn) ExecContext(ctx context.Context, query string, argsV []driver.NamedValue) (driver.Result, error) {
+	if !c.conn.IsConnected() {
+		return nil, driver.ErrBadConn
+	}
+
+	md := metadata.Pairs("authorization", c.Token)
+	ctx = metadata.NewOutgoingContext(ctx, md)
+
+	vals, err := namedValuesToSqlMap(argsV)
+	if err != nil {
+		return nil, err
+	}
+
+	execResult, err := c.conn.SQLExec(ctx, query, vals)
+	if err != nil {
+		return nil, err
+	}
+	return driver.RowsAffected(execResult.UpdatedRows), err
+}
+
+func (c *Conn) QueryContext(ctx context.Context, query string, argsV []driver.NamedValue) (driver.Rows, error) {
+	if !c.conn.IsConnected() {
+		return nil, driver.ErrBadConn
+	}
+	md := metadata.Pairs("authorization", c.Token)
+	ctx = metadata.NewOutgoingContext(ctx, md)
+
+	vals, err := namedValuesToSqlMap(argsV)
+	if err != nil {
+		return nil, err
+	}
+
+	queryResult, err := c.conn.SQLQuery(ctx, query, vals, true)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Rows{conn: c, rows: queryResult.Rows}, nil
+}
+
+func (c *Conn) Ping(ctx context.Context) error {
+	if !c.conn.IsConnected() {
+		return driver.ErrBadConn
+	}
+	err := c.conn.HealthCheck(ctx)
+	if err != nil {
+		c.Close()
+		return driver.ErrBadConn
+	}
+	return nil
+}
+
+func (c *Conn) CheckNamedValue(nv *driver.NamedValue) error {
+	// driver.Valuer interface is used instead
+	return nil
+}
+
+func (c *Conn) ResetSession(ctx context.Context) error {
+	if !c.conn.IsConnected() {
+		return driver.ErrBadConn
+	}
+	return ErrNotImplemented
+}
+
+type Rows struct {
+	index uint64
+	conn  *Conn
+	rows  []*schema.Row
+}
+
+func (r *Rows) Columns() []string {
+	if len(r.rows) > 0 {
+		names := make([]string, 0)
+		for _, n := range r.rows[0].Columns {
+			name := n[strings.LastIndex(n, ".")+1 : len(n)-1]
```

Comment:
I'm wondering if this is needed - is it required fro the name to not contain dot or be equal to column name?

Reply:
In immudb columns name are in the format {db}.{table}.{column-name}. From what I understood only {column-name is required}

Reply:
We can always add some flag in the future that will switch to full names if someone needs it